dangerzone/gui/main_window.py

```diff
@@ -501,17 +526,24 @@ def start_button_clicked(self) -> None:
                 (_, tmp) = tempfile.mkstemp(suffix=".pdf", prefix="dangerzone_")
                 document.output_filename = tmp
         else:
-            for document in self.dangerzone.get_unconverted_documents():
-                document.suffix = self.safe_extension.text()
+            if self.radio_move_untrusted.isChecked():
+                for document in self.dangerzone.get_unconverted_documents():
+                    document.archive_after_conversion()
+                    document.suffix = self.safe_extension.text()
 
-                if self.output_dir:
-                    document.set_output_dir(self.output_dir)  # type: ignore [unreachable]
+            elif self.radio_save_to.isChecked():
+                for document in self.dangerzone.get_unconverted_documents():
+                    document.suffix = self.safe_extension.text()
+
+                    if self.output_dir:
+                        document.set_output_dir(self.output_dir)  # type: ignore [unreachable]
```

Comment:
There are two things that bug me here:

1. Why is this operation a "this or that" thing. Can't we both archive the original document, and move the safe document to another directory? I don't get what's the conflict here.
2. There are some copy-pasted for-loops and function calls, so I've pushed a proposal that attempts to eliminate them.

Reply:
> There are some copy-pasted for-loops and function calls, so I've pushed a proposal that attempts to eliminate them.

Yes. There were some annoying duplicated code. I opted for having non-optimal iterations over all documents and duplicating the code. But your approach is better in the long run probably.

>  Why is this operation a "this or that" thing. Can't we both archive the original document, and move the safe document to another directory? I don't get what's the conflict here.

It doesn't have to be mutually exclusive. But it's two different ways of using the program. What would be the case if you archive all documents under the "unsafe" directory and at the same time move the safe files elsewhere? 

If there is a strong case for using it that way, then it makes sense. Otherwise, I'd push for a simpler usage, given that we're already increasing the UX (cognitive) complexity. Having 4 options instead of 2 might makes it more complex to the user. Maybe I'm exaggerating, but overall my point is the more use-case driven the UX is, the better (that won't make power users happy, though).

Reply:
Ok, makes sense then.